vault/data_source_transit_cmac_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+var cmacBlocks = `
+data "vault_transit_cmac" "test" {
+    path        = vault_mount.test.path
+    name        = vault_transit_secret_backend_key.test.name
+	input       = "aGVsbG8gd29ybGQuCg=="
+}
+data "vault_transit_verify" "test" {
+    path        = vault_mount.test.path
+    name        = vault_transit_secret_backend_key.test.name
+	input       = "aGVsbG8gd29ybGQuCg=="
+    cmac        = data.vault_transit_cmac.test.cmac
+}
+`
+
+var cmacBatchInputBlocks = `
+data "vault_transit_cmac" "test" {
+    path        = vault_mount.test.path
+    name        = vault_transit_secret_backend_key.test.name
+	batch_input = [
+		{
+		  input = "adba32=="
```

Comment:
Please consider adding a test with batch input that uses `reference`, and maybe `mac_length` (the Transit API docs example seems to indicate that that is possible).